### PR TITLE
Set up linting via pre-commit

### PR DESCRIPTION
**Pull request by akx**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - id: flynt
+  - repo: https://github.com/psf/black
+    rev: 22.10.0
+    hooks:
+      - id: black
+  - repo: https://github.com/PyCQA/flake8
+    rev: 5.0.4
+    hooks:
+      - id: flake8
+        additional_dependencies: &flake8_dependencies
```
Comment (ruksi) [inline .pre-commit-config.yaml:21]: anchors and aliases, oh my, the YAMLness 🥰

File: README.md [hunk 2]
```diff
@@ -274,8 +274,9 @@ Will produce this `valohai.yaml` config:
 
 If you wish to further develop `valohai-utils`, remember to install development dependencies and write tests for your additions.
 
+Linting is done via `pre-commit`, so be sure to set that up.
```
Comment (ruksi) [inline README.md:277]: we could be more helpful by specifying how to set it up :P 

e.g.

```
pip install pre-commit
pre-commit install
```

also, could mention that lints can be manually ran with something like `pre-commit run --all-files` (or `make lint` if you add that target to Makefile)

File: Makefile [hunk 3]
```diff
@@ -1,9 +0,0 @@
-prettify:
-	black .
-	isort .
-
-lint:
-	flake8 .
-	black --check .
-	isort -c .
-	mypy --strict .
```
Comment (ruksi) [inline Makefile:9]: could redefine as `pre-commit run --all-files` but I mean, not a must

**Review by ruksi — CHANGES_REQUESTED** [review]
changes LGTM, minor discussion points on the README

File: Makefile (hunk 3 above)
Comment (akx) [reply Makefile:9]: It's not exactly the same though, as the pre-commit things will change files...

Comment (ruksi) [reply Makefile:9]: yes, there is a slight difference, but I do prefer running it on everything if I do it manually while developing

but that is ofc only a personal preference :person_bald: 

we can have it without the `--all-files` too

Comment (akx) [reply Makefile:9]: I have an alias `pcr`, which does `pre-commit run --from-ref origin/$(git kit main-branch) --to-ref HEAD` (i.e. fixes up the branch I'm working on) (and could have `pca` for `--all-files`)...

Comment (ruksi) [reply Makefile:9]: those are definitely handy if you've got slower lints ! 🤩

Comment (ruksi) [reply Makefile:9]: no need to add the `make lint`, the README mention is good enough

**Review by ruksi — APPROVED** [review]

**Pull request merged**